src/main/java/com/sprint/mission/discodeit/service/basic/BasicChannelService.java

```diff
@@ -1,165 +1,124 @@
-    public List<Message> getMessagesFromChannel(UUID chId) {
-        return ChannelList.get(chId).getChannelMessageList();
-    }
-    @Override
-    public Map<UUID, Channel> getChannelList() {
-        return ChannelList;
+        return createdChannel;
     }
 
-    public void setChannelList(Map<UUID, Channel> channelList) {
-        this.ChannelList = channelList;
-    }
-    @Override
-    public void createChannel(Channel channel) {
-        ChannelList.put(channel.getId(), channel); //JCFChannelList update
-        participateChannel(channel.getOwner(), channel.getId());
-        //participate's == below two lines
-//        ChannelList.get(channel.getId()).getChannelUserList().add(channel.getOwner()); //channel's attending userList update
-//        JCFUserService.getInstance().getUserList().get(channel.getOwner()).getAttending().add(channel.getId()); //owner attending
-    }
     @Override
-    public Channel readChannelInfo(UUID id) {
-        Channel ch = ChannelList.get(id);
-        return ch;
+    public ChannelDto readChannelById(UUID channelId) {
+        return channelRepository.findById(channelId)
+                .map(this::toDto)
+                .orElseThrow(() -> new NoSuchElementException(channelId + " not found"));
     }
-    @Override
-    public <T> void updateChannelField(UUID channelId, String fieldName, T contents) {
-        String filePath = "/Users/kimsangho/Desktop/sprintMission/1-sprint-mission/src/main/java/com/sprint/mission/discodeit/properties/UpdatableChannelField.txt";
-        //String filePath = "../../properties/UpdatableChannelField.txt";
-        Channel channel = ChannelList.get(channelId);
-        if (channel == null) {
-            System.out.println("Channel doesn't exist!");
-        }
-        try (BufferedReader br = new BufferedReader(new FileReader(filePath))) {
-            String existingFieldName;
-            while ((existingFieldName = br.readLine()) != null) {
-                if (existingFieldName.equals(fieldName)) {
-                    String setterMethodName = "set" + Character.toUpperCase(fieldName.charAt(0)) + fieldName.substring(1);
-                    Method setterMethod = channel.getClass().getMethod(setterMethodName, contents.getClass());
-                    setterMethod.invoke(channel, contents);
-                    channel.setUpdatedAt();
-                    break;
-                }
 
+    @Override
+    public List<ChannelDto> readAllByUserId(UUID userId) {
+        List<UUID> mySubscribedChannelIds = readStatusRepository.findAllByUserId(userId).stream()
+                .map(ReadStatus::getChannelId)
+                .toList();
 
+        List<ChannelDto> result = new ArrayList<>();
+        List<Channel> allChannels = channelRepository.readAllContents();
 
+        for (Channel channel : allChannels) {
+            if (channel.getType().equals(ChannelType.PUBLIC) || mySubscribedChannelIds.contains(channel.getId())) {
+                result.add(toDto(channel));
```

Comment:
Java Stream을 사용하면 result라는 지역변수를 만들 필요 없이 바로 불변인 객체를 return할 수 있을 것 같네요.  

```suggestion
        return channelRepository.readAllContents().stream()
                .filter(channel -> channel.getType().equals(ChannelType.PUBLIC)
                        || mySubscribedChannelIds.contains(channel.getId()))
                .map(this::toDto)
                .toList();
```